README.md

```diff
@@ -39,58 +43,615 @@ Use a `<script>` tag to reference the package. Using this method will register a
+
+#### `etvasApiKey`
+
+This is a public key we provide (and you can find it in your account in [Etvas Partner's Studio](https://partners.helloetvas.com)) which is used not only to uniquely identify your app (the public part) but also to watch, analyze and shape the incoming traffic to automatically protect against a wide range of network attacks. Please note the entire process is anonymous and no personal data is stored by these operations.
+
+#### `organizationId`
+
+This is an UUID that uniquely identifies your organization inside Etvas Ecosystem. Paired with the ApiKey, it provides a wide range of optimizations we provide for your network calls, in an ongoing effort to improve the response times for our API's.
+
+#### `locale`
+
+We support internationalization out of the box. Our I18N system will dynamically load the selected language dictionary and update the User Interface in an instant, providing for your customer the best experience available. We currently have dictionaries for `en` - English language and `de` - German language, with more coming.
+
+## The building blocks
+
+As mentioned before, Etvas Automat is designed as separate building blocks. This architecture allows you to take control over every rendered position, event and click the customer performs on the displayed content.
+
+There are a number of different sections that you can display with Etvas Automat. Some of them can be displayed directly, others require intermediary steps, like, for example, in order to display the Account Settings part, you need to authenticate the user through OpenID.
+
+We will describe the authentication mechanism first and, for each building block, we will note if it requires the authentication to be performed first or not.
+
+### Authentication (`connect` and `logout`)
+
+When displaying the authenticated content (`showMyProducts`, `showSettings`) or when the purchase flow must happen, you must connect the Etvas Platform to your OpenID Authentication server. This means you must call the `connect` function on Etvas Automat, which will open a small popup window that gives the user the means to login (or automatically register!!) in Etvas using your authentication server.
+
+**Please note** this flow requires additional setup both on Etvas Platform and your authentication server, which must support OpenID protocol.
+
+The signature is:
 
 ```
-etvasAutomat.showProductCard('.etvas-placeholders`)
+automat.connect(callback)
 ```
 
-Please note, if you are using jQuery, please use the following syntax, in order to have the native JS DOM element and not a jQuery wrapper:
+where `callback` is a function that will run when successfully connected.
+
+The function accepts a callback which will be called without any arguments but only on successful login:
 
 ```
-const $element = $('#etvas-placeholder')
-etvasAutomat.showProductCard($element.first()[0], options)
+automat.connect(function () {
```

Comment:
how do I know if there was an error on connect ?

Reply:
This is out of our hands, we can only open the window (it's managed by Cognito). We can only detect an auth, in which case the callback is called. No errors whatsoever.